File: old/application/classes/social/reddit.php
<?php defined('SYSPATH') or die('No direct script access.');

/**
 * Support for sharing posts on Reddit.
 */
class Social_Reddit extends Social implements Social_Share
{
	const SHARE_URL = 'http://reddit.com/submit';
	const API_URL = 'http://www.reddit.com/api/info.json';
	
	/**
	 * Get the URL to share this post
	 * @param	Model_Blog_Post		The post
	 */
	public function share_url(Model_Blog_Post $post)
	{
		return self::SHARE_URL . '?' . http_build_query(array(
			'url' => $post->url(true),
			'title' => $post->title,
		));
	}
	
	/**
	 * Get the number of times this URL has been shared
	 * @param	Model_Blog_Post		The post
	 */
	public function share_count(Model_Blog_Post $post)
	{
		$total = 0;
		
		$url = self::API_URL . '?' . http_build_query(array(
			'url' => $post->url(true),
		));
		
		$data = json_decode(file_get_contents($url));
		if (empty($data) || empty($data->data) || empty($data->data->children))
			return 0;
			
		// Need to add up the points in every submission of this URL
		foreach ($data->data->children as $child)
			$total += $child->data->score;
		
		return $total;
	}
}

?>

File: Daniel15.BusinessLayer/Services/Social/Reddit.cs
using System;
using System.Net;
using Daniel15.Data.Entities.Blog;
using Microsoft.AspNetCore.Http.Extensions;
using Newtonsoft.Json.Linq;

namespace Daniel15.BusinessLayer.Services.Social
{
	/// <summary>
	/// Support for sharing posts on Reddit
	/// </summary>
	public class Reddit : ISocialShare
	{
		/// <summary>
		/// Base URL for Reddit share URLs
		/// </summary>
		private const string SHARE_URL = "http://reddit.com/submit";
		/// <summary>
		/// URL to retrieve sharing count
		/// </summary>
		private const string API_URL = "http://www.reddit.com/api/info.json";

		/// <summary>
		/// Gets the internal ID of this social network
		/// </summary>
		public string Id => "reddit";

		/// <summary>
		/// Gets the friendly name of this social network
		/// </summary>
		public string Name => "Reddit";

		#region Implementation of ISocialShare
		/// <summary>
		/// Gets the URL to share this post on this social network.
		/// </summary>
		/// <param name="post">The blog post</param>
		/// <param name="url">Full URL to this post</param>
		/// <param name="shortUrl">Short URL to this post</param>
		/// <returns>Sharing URL for this post</returns>
		public string GetShareUrl(PostModel post, string url, string shortUrl)
		{
			return SHARE_URL + new QueryBuilder
			{
				{"url", url},
				{"title", post.Title}
			};
		}

		/// <summary>
		/// Gets the number of times this URL has been shared on this social network.
		/// </summary>
		/// <param name="post">The blog post</param>
		/// <param name="url">Full URL to this post</param>
		/// <param name="shortUrl">Short URL to this post</param>
		/// <returns>Share count for this post</returns>
		public int GetShareCount(PostModel post, string url, string shortUrl)
		{
			var total = 0;
			var apiUrl = API_URL + new QueryBuilder
			{
				{"url", url}
			};

			using (var client = new WebClient())
			{
				var rawData = client.DownloadString(apiUrl);
				dynamic data = JObject.Parse(rawData);
				if (data == null || data.data == null || data.data.children == null)
					return 0;

				// Need to add up the points in every submission of this URL
				foreach (var child in data.data.children)
				{
					total += Convert.ToInt32(child.data.score);
				}

				return total;
			}
		}
		#endregion
	}
}
